### ISSUE TITLE
App crashes on opening file picker on installing from TestFlight. Works fine when run directly from XCode

### ISSUE THREAD

**Issue by cerebroo** [jcesarmobile/FilePicker-Phonegap-iOS-Plugin#50]

**Comment by daveshirman** [comment]
Is this on an iOS 10.x device?

I've had the same issue yesterday with iOS 10.x, but works fine with iOS 11.x

https://stackoverflow.com/questions/50622920/ionic-3-ios-file-picker-crashing-app-on-device-running-ios-10-x-not-11

**Comment by cerebroo** [comment]
@daveshirman - happening on iOS 11.3

Strangely works fine when I run directly from xcode but fails when I push to TestFlight and install from there.

**Comment by tvarghesedt** [comment]
Same for me.. I created a diawi test url and when installing it crash when I click on choose file. Do anyone have solution?

**Comment by tvarghesedt** [comment]
@daveshirman Did you finally got a solution.. I am still stuck on this

**Comment by daveshirman** [comment]
@tvarghesedt sadly no.  For now (and the foreseeable future) I've put a device version check in the code and just disabled the feature for iOS 10.  Everything is fine on iOS 11.

**Comment by saanthoshg** [comment]
hi, 
you got any solution?
App is crashing for iOS11 as well.

**Comment by plindsay** [comment]
I've experienced this issue as well - works fine in Xcode and Emulator but crashes when installed on device via TestFlight. Very frustrating :-/

I've since swapped the integration to this plugin which seems to be more stable -
https://github.com/iampossible/Cordova-DocPicker

**Comment by inexuscore** [comment]
Has anyone got a solution for this? I've been struggling with this for weeks now and my client is agitated. I thought it has something to do with Xcode and my signing certificates, so I revoked and deleted everything, regenerated certificates and provisioning profiles etc. very strange. I cannot debug TestFlight builds to see what's happening, obviously. My app uses this plugin to allow attachments to be chosen from your iCloud drive. works fine when run directly from Xcode or in a simulator. But not in TestFlight builds. Nothing happens, no errors whatsoever I don't even know what's going wroing here. 

I'm testing my app on iPhone 8 + iOS 12 + Xcode 10. Any ideas? I need to solve this ASAP!

**Comment by inexuscore** [comment]
> I've experienced this issue as well - works fine in Xcode and Emulator but crashes when installed on device via TestFlight. Very frustrating :-/
> 
> I've since swapped the integration to this plugin which seems to be more stable -
> https://github.com/iampossible/Cordova-DocPicker

no documentation whatsoever. how do I use this plugin?

**Comment by plindsay** [comment]
Yeah, the Github page doesn't help much but there's info on the Ionic site -
https://ionicframework.com/docs/native/ios-documentpicker/

**Comment by inexuscore** [comment]
@plindsay thank you very much. It's great that we can use Ionic Native plugins in any cordova project. I'll give it a try. I have also found that using a File Input works too. Apparently it only works in iOS 11+ so it might not be a permanent solution. but looks good to me. it popups up a dialog with three options:

- Take a picture (opens camera)
- Pick from library
- Browse iCloud Drive 

I'll post my findings here if I had any luck. Cheers!

**Comment by inexuscore** [comment]
@plindsay Thanks a lot! The Ionic Native plugin works like a charm. Needs thorough testing though and it seems the list of supported files is limited. It has a 'ALL' option but I can't seem to pick .pages files. Anyway good for now, cheers!

**Comment by plindsay** [comment]
@inexuscore no problem - glad you got sorted! 😀
